rasa/utils/endpoints.py

```diff
@@ -96,6 +95,7 @@ def __init__(
         self.cafile = cafile
         self.kwargs = kwargs
 
+    @lru_cache
```

Comment:
Should we opt to use `cached_property` instead of `lru_cache`? According to [the docs](https://docs.python.org/3/faq/programming.html#faq-cache-method-calls), `lru_cache` is recommended when the method contains args, but this `session()` method doesn't. Also removes the need to implement `__hash__`.

Reply:
@ancalita I think cached_property would be better here as its precisely built for caching the class properties